src/Perception/include/BipedalLocomotion/Perception/Features/ArucoDetector.h

```diff
@@ -78,6 +78,29 @@ class ArucoDetector : public System::Source<ArucoDetectorOutput>
      * - "marker_length" marker length in m
      * - "camera_matrix" 9d vector representing the camera calbration matrix in row major order
      * - "distortion_coefficients" 5d vector containing camera distortion coefficients
+     * - "use_solve_pnp_outlier_rejection" solve for marker pose using PnP method instead of default
+     *                                     aruco pose estimation, this method allows to eliminate
+     *                                     outliers in estimated rotations, in the form of rotation flips,
+     *                                     upto a certain degree by considering ambiguous solutions as outliers
+     *                                     and removing associated detections from the set of detected markers
+     *                                     if using this option, please cite the work,
+     *                                     "Infinitesimal Plane-Based Pose Estimation"
+     *                                     (https://link.springer.com/article/10.1007/s11263-014-0725-5)
+     *                                     and also cite,
+     *                                     "Absolute humanoid localization and mapping based on IMU
+     *                                      Lie group and fiducial markers"
+     *                                     (https://ieeexplore.ieee.org/document/9035005)
+     *
+     * - "ambiguity_threshold_reprojection_error_ratio" outlier rejection using solvePnP method is done based on
+     *                                                  reprojection error ratio between the obtained solutions.
+     *                                                  threshold ratio of reprojection error for deciding ambiguous rotations
+     *                                                  pose estimation for planar surfaces (homography) usually result in
+     *                                                  two rotation solutions (with one-solution having z axis flipped in opposite)
+     *                                                  the rotation with a lower reprojection error is a suitable solution
+     *                                                  we use a reprojection error ratio test to decide if two rotation solutions
+     *                                                  are close to each other, higher ratio means distinct rotations
+     *                                                  if the resulting ratio is small, then we mark the estimated pose as an outlier
+     *                                                  higher the value, low frequent becomes successful marker detection
```

Comment:
Can you please rephrase it a little? I had a bit of a hard time understanding this

Reply:
Fixed in https://github.com/ami-iit/bipedal-locomotion-framework/pull/503/commits/b1822e82a41d7c63b27f79ec7d81e16e721a0be9.